### PR TITLE
ci(cypress): Add bank redirects for stripe

### PR DESCRIPTION
**Pull request by likhinbopanna**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [x] CI/CD

## Description
Add Bank redirects payment methods for stripe in cypress automation

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
Bank redirects  test cases was present in the automation setup

## How did you test it?
Cypress 
- Stripe
image

- Adyen
image

- Trustpay
image

- Nmi
image
 
- Cybersource
image

- Bankofamerica
image

- Bluesnap
image

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: cypress-tests/cypress/e2e/PaymentTest/00017-BankRedirect.cy.js [hunk 1]
```diff
@@ -301,7 +301,67 @@ describe("Bank Redirect tests", () => {
...
+        true,
+        globalState
+      );
+      if (should_continue)
+        should_continue = utils.should_continue_further(res_data);
+    });
+
+    it("Handle bank redirect redirection", () => {
+      let expected_redirection = confirmBody["return_url"];
+      let payment_method_type = confirmBody["payment_method_type"];
```
[the remark this answers is not in the document]
Comment (likhinbopanna) [reply cypress-tests/cypress/e2e/PaymentTest/00017-BankRedirect.cy.js:360]: - Payout test cases are not handled yet
- Will take it as a separate PR

File: cypress-tests/cypress/e2e/PaymentUtils/Stripe.js [hunk 2]
```diff
@@ -370,4 +372,141 @@ export const connectorDetails = {
...
+      Request: {
+        payment_method: "bank_redirect",
+        payment_method_type: "giropay",
+        payment_method_data: {
+          bank_redirect: {
+            giropay: {
+              bank_name: "",
+              bank_account_bic: "",
+              bank_account_iban: "",
+              preferred_language: "en",
```
[the remark this answers is not in the document]
Comment (likhinbopanna) [reply cypress-tests/cypress/e2e/PaymentUtils/Stripe.js]: done

File: cypress-tests/cypress/e2e/PaymentUtils/Stripe.js [hunk 3]
```diff
@@ -370,4 +372,141 @@ export const connectorDetails = {
...
+              blik_code: "777987",
+            },
+          },
+        },
+      },
+      Response: {
+        status: 200,
+        body: {
+          status: "failed",
+          error_code: "payment_intent_invalid_parameter",
```
[the remark this answers is not in the document]
Comment (likhinbopanna) [reply cypress-tests/cypress/e2e/PaymentUtils/Stripe.js:481]: done

File: cypress-tests/cypress/support/redirectionHandler.js [hunk 4]
```diff
@@ -254,6 +254,31 @@ function bankRedirectRedirection(
...
+        case "przelewy24":
+          cy.get('a[name="success"]').click();
+          break;
+        default:
+          throw new Error(
+            `Unsupported payment method type: ${payment_method_type}`
+          );
+      }
+      verifyUrl = true;
+      break;
```
[the remark this answers is not in the document]
Comment (likhinbopanna) [reply cypress-tests/cypress/support/redirectionHandler.js]: done

File: cypress-tests/cypress/e2e/PaymentTest/00017-BankRedirect.cy.js [hunk 5]
```diff
@@ -301,7 +301,67 @@ describe("Bank Redirect tests", () => {
...
+        "three_ds",
+        "automatic",
+        globalState
+      );
+      if (should_continue)
+        should_continue = utils.should_continue_further(res_data);
+    });
+
+    it("payment_methods-call-test", () => {
+      cy.task("cli_log", "PM CALL ");
```
Comment (pixincreate) [inline cypress-tests/cypress/e2e/PaymentTest/00017-BankRedirect.cy.js:336]: we can remove this line. not needed.

File: cypress-tests/cypress/e2e/PaymentTest/00017-BankRedirect.cy.js [hunk 6]
```diff
@@ -301,7 +301,67 @@ describe("Bank Redirect tests", () => {
...
+      cy.paymentMethodsCallTest(globalState);
+    });
+
+    it("Confirm bank redirect", () => {
+      let data = getConnectorDetails(globalState.get("connectorId"))[
+        "bank_redirect_pm"
+      ]["przelewy24"];
+      let req_data = data["Request"];
+      let res_data = data["Response"];
+      cy.task("cli_log", "GLOBAL STATE -> " + JSON.stringify(globalState.data));
```
Comment (pixincreate) [inline cypress-tests/cypress/e2e/PaymentTest/00017-BankRedirect.cy.js:346]: remove this

**Review by pixincreate — COMMENTED** [review]
Other than that LGTM.

File: cypress-tests/cypress/e2e/PaymentUtils/Adyen.js [hunk 7]
```diff
@@ -437,11 +437,7 @@ export const connectorDetails = {
         payment_method_type: "giropay",
         payment_method_data: {
           bank_redirect: {
-            giropay: {
-              bank_name: "",
-              bank_account_bic: "",
-              bank_account_iban: "",
```
[the remark this answers is not in the document]
Comment (likhinbopanna) [reply cypress-tests/cypress/e2e/PaymentUtils/Adyen.js]: sure

File: cypress-tests/cypress/e2e/PaymentTest/00017-BankRedirect.cy.js (hunk 5 above)
Comment (likhinbopanna) [reply cypress-tests/cypress/e2e/PaymentTest/00017-BankRedirect.cy.js]: done

**Review by pixincreate — APPROVED** [review]

**Review by Gnanasundari24 — APPROVED** [review]

**Pull request merged**